src/opendr/perception/object_detection_2d/gem/gem_learner.py

```diff
@@ -0,0 +1,1059 @@
+            m1_data_loader, m2_data_loader, base_ds, device,
+            self.temp_path, verbose=verbose
+        )
+
+        return test_stats
+
+    def infer(self, m1_image, m2_image):
+        """
+        This method is used to perform object detection on an image.
+
+        Parameters
+        ----------
+        image : engine.data.Image class object
+            Image to run inference on.
+
+        Returns
+        -------
+        engine.target.BoundingBoxList
+            The engine.target.BoundingBoxList contains bounding boxes that are
+            described by the left-top corner and its width and height, or
+            returns an empty list if no detections were made.
+
+        """
+        if not (isinstance(m1_image, Image) or isinstance(m2_image, Image)):
+            m1_image = Image(m1_image)
+            m2_image = Image(m2_image)
+
+        m1_img = im.fromarray(m1_image.numpy())
+        m2_img = im.fromarray(m2_image.numpy())
+
+        scores, boxes, segmentations, contrib = detect(m1_img, m2_img, self.infer_transform, self.model,
+                                                       self.postprocessors, self.device, self.threshold,
+                                                       self.ort_session,
+                                                       )
+        weight1 = contrib[0].data
+        weight1 = weight1.cpu().detach().numpy()
+
+        weight2 = contrib[1].data
+        weight2 = weight2.cpu().detach().numpy()
+
+        normed_weight1 = weight1 / (weight1 + weight2)
+        normed_weight2 = weight2 / (weight1 + weight2)
+
+        boxlist = []
+        if len(segmentations) == len(scores):
+            for p, (xmin, ymin, xmax, ymax), segmentation in zip(scores.tolist(), boxes.tolist(), segmentations):
+                cl = np.argmax(p)
+                box = CocoBoundingBox(cl, xmin, ymin, xmax - xmin, ymax - ymin, score=p[cl], segmentation=segmentation)
+                boxlist.append(box)
+        else:
+            for p, (xmin, ymin, xmax, ymax) in zip(scores.tolist(), boxes.tolist()):
+                cl = np.argmax(p)
+                box = CocoBoundingBox(cl, xmin, ymin, xmax - xmin, ymax - ymin, score=p[cl])
+                boxlist.append(box)
+        return BoundingBoxList(boxlist), normed_weight1, normed_weight2
+
+    def optimize(self, do_constant_folding=False):
+        """
+        Method for optimizing the model with onnx. (Not implemented yet for GEM)
+
+        Parameters
+        ----------
+        do_constant_folding : bool, optional
+            If true, constant folding is true in the onnnx model. The default
+            is False.
+
+        Raises
+        ------
+        UserWarning
+            If no model is loaded or if an ort session is already ongoing, a
+            user warning is raised.
+
+        Returns
+        -------
+        bool
+            True if the model was optimized successfully.
+
+        """
```

Comment:
Please raise a warning, since this method is not implemented.

Reply:
Thank you for pointing this out, I will add it to the learner.